### PR TITLE
Add env variable for nvidia pytorch wheel release automation.

### PR DESCRIPTION
**Pull request by nWEIdia**
Background: NVIDIA releases monthly NGC containers based on pytorch, e.g. https://docs.nvidia.com/deeplearning/frameworks/pytorch-release-notes/rel-24-03.html 
This PR is needed to facilitate the above efforts, avoiding the need to cherry pick this change every month. 

cc @malfet @atalman @ptrblck

### CODE REVIEW

**Comment by ptrblck** [comment]
@nWEIdia would it make more sense to allow more general flexibility in naming the wheel instead of NVIDIA-specifc?

**Comment by nWEIdia** [comment]
> @nWEIdia would it make more sense to allow more general flexibility in naming the wheel instead of NVIDIA-specifc?

It might be possible to generalize the environment variables like APPLE_PYTORCH_VERSION, INTEL_PYTORCH_VERSION, AMD_PYTORCH_VERSION to COMPANY_PYTORCH_VERSION, etc, but the subsequent YY.MM might be too specific for us that others will not find that useful.

File: tools/generate_torch_version.py [hunk 1]
```diff
@@ -60,6 +60,15 @@ def get_torch_version(sha: Optional[str] = None) -> str:
         if sha is None:
             sha = get_sha(pytorch_root)
         version += "+git" + sha[:7]
+    if os.getenv("NVIDIA_PYTORCH_VERSION"):
+        # see if it matches YY.MM
+        str_nvidia_pytorch_version = os.getenv("NVIDIA_PYTORCH_VERSION")
```
Comment (malfet) [inline tools/generate_torch_version.py:65]:
```suggestion
    if str_nvidia_pytorch_version := os.getenv("NVIDIA_PYTORCH_VERSION"):
        # see if it matches YY.MM
```

File: tools/generate_torch_version.py [hunk 2]
```diff
@@ -60,6 +60,15 @@ def get_torch_version(sha: Optional[str] = None) -> str:
...
+        pattern = re.compile(r'^\d{2}\.\d{2}$')
+        if bool(pattern.match(str_nvidia_pytorch_version)):
+          version += ".nv" + str_nvidia_pytorch_version
+        else:
+          version += ".nvInternal"
```
Comment (malfet) [inline tools/generate_torch_version.py:70]: How about passing it explicitly via `PYTORCH_VERSION_SUFFIX`

Comment (nWEIdia) [reply tools/generate_torch_version.py:69]: As discussed, I will think more about it and re-open when it is ready.

**Comment by nWEIdia** [comment]
Need re-work, closing for now

**Pull request closed without merging**